packages/gestalt/src/Popover.js

```diff
@@ -108,15 +155,39 @@ export default function Popover({
       id={id}
       idealDirection={idealDirection}
       onDismiss={onDismiss}
-      positionRelativeToAnchor={positionRelativeToAnchor}
+      disablePortal={disablePortal ?? positionRelativeToAnchor}
       color={color === 'deprecatedBlue' ? 'blue' : color}
       role={role}
       shouldFocus={shouldFocus}
       showCaret={_deprecatedShowCaret}
       size={size}
-      __dangerouslySetMaxHeight={__dangerouslySetMaxHeight}
+      scrollBoundary={scrollBoundary}
+      hideWhenReferenceHidden={hideWhenReferenceHidden}
     >
       {children}
     </InternalPopover>
   );
 }
+
+export function ExperimentalPopover({
```

Comment:
The ExperimentProvider goes in Pinboard. We read from within the component, but we set the provider at the app level

Reply:
Yes. The experiments for Popover and its dependent components are separate. 

For example, ComboBox has its own experiment and Popover has its own too. When ComboBox experiment is switched on, it has to switch on the Popover experiment for the underlying Popover.

I could just add a new prop directly to Popover but it would pollute the API.

Reply:
Oh I see. Just add a prop to Popover _experimentalPopover, an internal prop basically rather than triggering the experiment. We've done this in the past. https://gestalt.pinterest.systems/get_started/developers/contributing/experimentation#Experimental-prop 

Here you won't be able to set (easily) 2 experiments that simultaneously get the same audience in the experiment. 

Let's control the experiment from the dependent components and have them internally trigger the change in Popover.

Reply:
If I understand you correcly, internally we should control the experiment via `_experimentalPopover` prop and not use `ExperimentProvider`. And in Pinboard we will use `ExperimentProvider` to control the experiment. Right?

Reply:
Yes
in Pinboard we will use ExperimentProvider to control 

- ComboBox
  - web_gestalt_popover_v2_combobox
  - mweb_gestalt_popover_v2_combobox
- Dropdown
  - web_gestalt_popover_v2_dropdown
  - mweb_gestalt_popover_v2_dropdown
- HelpButton
  - web_gestalt_popover_v2_helpbutton
  - mweb_gestalt_popover_v2_helpbutton
- OverlayPanel
  - web_gestalt_popover_v2_confirmationpopover
  - web_gestalt_popover_v2_confirmationpopover
- PopoverEducational
  - web_gestalt_popover_v2_popovereducational
  - web_gestalt_popover_v2_popovereducational

Then internally, we pass a hidden prop from each component to Popover via _experimentalPopover communicating that Popover is in the experiment

Reply:
This will allow you to "also" test Popover manually on individual instances in Pinboard. You can do _experimentalPopover={true} in a Popover and see what happens.